### PR TITLE
Amount component

### PR DESCRIPTION
**Pull request by futurepaul**
# Description

A `<Value />` component for displaying values throughout the app.

This needs some cleanup and would love to chat if there's anything I'm doing here that could be simpler / more straightforward.

Also it should be pretty straighforward to rewrite the existing `getAmount` method based on the new logic of `getUnformattedAmount` method but since I believe we can deprecate `getAmount` eventually, so not sure if that's necessary.

Another thing this component could be set up to support is displaying bytes / vbytes. Didn't bother with that for now but should be easy addition.

Also, in addition to various rules about left / right and spacing for monies, there are also rules about commas:

https://fastspring.com/blog/how-to-format-30-currencies-from-countries-all-over-the-world/

For instance here's Argentina:
`$ 1.234,56`
And here's USA:
`$1,234.56`

Shouldn't be too hard to add but haven't done that here.

I'd love some recommendations on how to kick the tires of this to make sure it's truly robust.

Finally, this won't every look like the mockups in Figma until we get Lato installed :)

image
image
image
image

This pull request is categorized as a:

- [x] New feature
- [ ] Bug fix
- [x] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Other

## Checklist
- [ ] I’ve run `npm run tsc` and make sure my code didn’t produce errors 
- [ ] I’ve run `npm run prettier` and formatted my code correctly

## Testing

If you added new functionality or fixed a bug, did you add new unit tests?

- [x] No, I’m a fool
- [ ] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [x] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] LND
- [ ] c-lightning-REST
- [ ] spark
- [ ] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `npm install` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `package-lock.json` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

**Comment by futurepaul** [comment]
We talked and we're going to change this component name to `Amount`.

File: stores/UnitsStore.ts [hunk 1]
```diff
@@ -47,6 +57,65 @@ export default class UnitsStore {
...
 
+    @action getUnformattedAmount = (
+        value: string | number = 0,
+        fixedUnits?: string
+    ): ValueDisplayProps => {
+        const { settings } = this.settingsStore;
+        const { fiat } = settings;
+        const units = fixedUnits || this.units;
+
+        const wholeSats = value.toString().split('.')[0];
```
Comment (kaloudis) [inline stores/UnitsStore.ts:68]: `const [wholeSats] = value.toString().split('.');` maybe

File: views/Wallet/MainPane.tsx [hunk 2]
```diff
@@ -53,71 +54,23 @@ export default class MainPane extends React.PureComponent<MainPaneProps, {}> {
...
-                >
-                    {units &&
-                        PrivacyUtils.sensitiveValue(
-                            getAmount(lightningBalance),
-                            8,
-                            true
-                        )}{' '}
-                    ⚡
-                </Text>
+                <Value jumbo={true} sats={lightningBalance} sensitive />
```
Comment (futurepaul) [inline views/Wallet/MainPane.tsx:57]: pass the jumbo prop like the sensitive prop

**Comment by futurepaul** [comment]
Okay this should be good to go sorry for the delay.

I tried rebasing this to master and there were no conflicts which is amazing to me. Let me know if I should push that version up and point this PR at master.

**Comment by kaloudis** [comment]
@futurepaul awesome, please do

File: components/LayerBalances/index.tsx [hunk 3]
```diff
@@ -80,29 +75,22 @@ const SwipeableRow = ({
     );
 };
 
-@inject('UnitsStore')
+@inject()
```
Comment (kaloudis) [inline components/LayerBalances/index.tsx:78]: do you still need this `inject` here if it's empty?

**Comment by kaloudis** [comment]
Can we add a color prop to amount? Some balances will be red, green, or the highlight themeColor

**Comment by ZeusLN** [comment]
Merging this in as is so we can start integrating it other places.

We can iterate on it in future PRs.

**Pull request merged**